src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -161,10 +169,16 @@ package object csv {
 
           override def next: String = {
             if (iter.nonEmpty) {
-              val row = csvFormat.format(iter.next().toSeq.map(_.asInstanceOf[AnyRef]): _*)
+              val values: Seq[AnyRef] = iter.next().toSeq.map(
```

Comment:
Do have a sense of the overhead of this additional type checking for each record? If the overhead is substantial, maybe we can check the schema for existence of `TimestampType` and/or `DateType` and if we find one, then we can perform the proper formatting. Even better would be using the index of the field instead of type check when we know there are Date or Time types.